packages/trpc/server/routers/viewer/attributes/edit.handler.ts

```diff
@@ -109,6 +114,29 @@ const editAttributesHandler = async ({ input, ctx }: GetOptions) => {
   return attributes;
 };
 
+/**
+ * Ensures that contains has no non-existent sub-options
+ */
+function getOptionsWithValidContains(options: ZEditAttributeSchema["options"]) {
+  return options.map(({ contains, ...option }) => {
+    if (!contains)
+      return {
+        ...option,
+        contains: [],
+      };
+    const possibleSubOptions = options
+      .filter((option) => !option.isGroup)
+      .filter((option): option is typeof option & { id: string } => option.id !== undefined);
```

Comment:
I'm not a fan of calling array methods back to back. Instead we should be using `.reduce()`

Reply:
I actually prefer filter chaining as it keeps the code more declarative and easier to read. 'reduce' takes an additional mental effort to unpack what it is doing.

I would drop this here
![image](https://github.com/user-attachments/assets/7e9c1378-22e7-44e1-95f5-0b1193545a9f)


Source: https://www.davideasaf.com/mapfilter-vs-reduce/

Reply:
Thanks for sharing the blog post @hariombalhara. I understand where you're coming from. You're right these aren't going to be huge arrays so these won't be a blocker for me.